lib/modules/manager/cpanfile/extract.spec.ts

```diff
@@ -0,0 +1,252 @@
+import { extractPackageFile } from '.';
+
+describe('modules/manager/cpanfile/extract', () => {
+  describe('extractPackageFile()', () => {
+    it('returns null for empty', async () => {
+      expect(await extractPackageFile('nothing here', 'cpanfile')).toBeNull();
+    });
+
+    it('parse perl', async () => {
+      expect(
+        await extractPackageFile(`requires 'perl', '5.008001';`, 'cpanfile')
+      ).toEqual({
+        deps: [
+          {
+            versioning: 'perl',
+            depName: 'Perl',
+            packageName: 'Perl/perl5',
+            currentValue: '5.008001',
+            datasource: 'github-tags',
+          },
+        ],
+      });
+    });
+
+    it('parse modules with requires', async () => {
+      expect(
+        await extractPackageFile(
+          `requires 'Try::Tiny';
```

Comment:
Please, search in our codebase for `codeBlock` usage, it's great for multi-line source embedding